modules/highgui/src/window.cpp

```diff
@@ -1082,6 +1082,42 @@ void cv::imshow(const String& winname, const ogl::Texture2D& _tex)
 #endif
 }
 
+const std::string cv::currentUIFramework()
+{
+#if defined(OPENCV_HIGHGUI_WITHOUT_BUILTIN_BACKEND) && defined(ENABLE_PLUGINS)
+    auto backend = getCurrentUIBackend();
+    if (backend)
+    {
+        return backend->getName();
+    }
+    CV_Error(Error::StsNotImplemented, "The function is not implemented with this GUI. ");
+    return std::string();
+#elif defined(HAVE_WIN32UI)
+    auto backend = getCurrentUIBackend();
```

Comment:
What is the problem with scheme similar to `setWindowTitle`?

Only one branch should try "plugin" interface. Other are builtins and they are pre-defined constants.

Reply:
I hope I understang plugins well